gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/hive/HivePartitionFileSet.java

```diff
@@ -99,7 +99,7 @@ public HivePartitionFileSet(HiveCopyEntityHelper hiveCopyEntityHelper, Partition
               hiveCopyEntityHelper.getExistingEntityPolicy() != HiveCopyEntityHelper.ExistingEntityPolicy.REPLACE_TABLE_AND_PARTITIONS) {
             log.error("Source and target partitions are not compatible. Aborting copy of partition " + this.partition,
                 ioe);
-            return Lists.newArrayList();
+            throw ioe;
```

Comment:
Right now there isn't, but we can explore adding more fine-tuned control if this ends up being too noisy. I think since we can also control whether to fail loudly or not with an override policy it's better to throw here than leave silent failures since it breaks user retry capability.

Reply:
"control whether to fail loudly or not with an override policy" this is the configure I was referring, if we can do that already, this change is much low-risk than I thought. thanks for clarification

Reply:
Does the override policy already exist @Will-Lo ?
If there are 10 partitioned tables to be copied and the 3rd table has incompatible target location, will 2 tables be copied or 9 or none?

Reply:
Right now it would be no tables, unless the user specifies the override policies to overwrite their partition locations to these new locations.